### PR TITLE
Refactored exit signature rotation

### PR DESCRIPTION
**Pull request by cyc60**

### CODE REVIEW

File: src/common/utils.py [hunk 1]
```diff
@@ -44,21 +42,11 @@ def log_verbose(e: Exception):
...
-        await asyncio.sleep(sleep_time)
-        start = time.time()
-
-        chain_head = await consensus_client.get_chain_finalized_head(
-            settings.network_config.SLOTS_PER_EPOCH
-        )
-
-        elapsed = time.time() - start
-        sleep_time = float(settings.network_config.SECONDS_PER_BLOCK) - elapsed
+async def is_block_synced(block_number: BlockNumber):
```
Comment (tsudmi) [inline src/common/utils.py:45]: missing return type

File: src/exits/tasks.py [hunk 2]
```diff
@@ -39,99 +33,39 @@ async def update_exit_signatures_periodically(
...
+    endpoints = [endpoint for replicas in oracles.endpoints for endpoint in replicas]
+    shuffle(endpoints)
+
+    for oracle_endpoint in endpoints:
+        response = await get_oracle_outdated_signatures_response(oracle_endpoint)
+        if not update_block or response['exit_signature_block_number'] >= update_block:
+            outdated_indexes = [val['index'] for val in response['validators']]
+            metrics.outdated_signatures.set(len(outdated_indexes))
+            return outdated_indexes
+    raise RuntimeError('Oracles are not synced exit signatures, waiting...')
```
Comment (evgeny-stakewise) [inline src/exits/tasks.py:68]: No waiting here. Log message needs to be corrected.

Comment (evgeny-stakewise) [inline src/exits/tasks.py:68]:
```suggestion
    raise RuntimeError('Oracles have not synced exit signatures yet')
```

File: src/exits/tasks.py [hunk 3]
```diff
@@ -39,99 +33,39 @@ async def update_exit_signatures_periodically(
...
-    metrics.outdated_signatures.set(len(outdated_indexes))
-    return outdated_indexes
-
-
-async def _wait_oracles_signature_update(oracles: Oracles) -> None:
-    last_event = await keeper_contract.get_exit_signatures_updated_event(vault=settings.vault)
-    if not last_event:
+    update_block = await _fetch_last_update_block()
+    if update_block and not await is_block_synced(update_block):
+        logger.info('Waiting for block %d finalization...', update_block)
```
Comment (evgeny-stakewise) [inline src/exits/tasks.py:39]:
```suggestion
        logger.info('Signatures update block %d is not finalized yet', update_block)
```

**Review by evgeny-stakewise — APPROVED** [review]

**Pull request merged**